File: AOJ/ITP1/itp_1_4_d.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

var sc = bufio.NewScanner(os.Stdin)

func init() {
	sc.Split(bufio.ScanWords)
}

func readInt() int {
	sc.Scan()
	r, _ := strconv.Atoi(sc.Text())
	return r
}

func main() {
	n := readInt()
	arr := []int{}
	for i := 0; i < n; i++ {
		arr = append(arr, readInt())
	}
	max, min, sum := -1000000, 1000000, 0
	for _, v := range arr {
		if max < v {
			max = v
		}
		if v < min {
			min = v
		}
		sum += v
	}
	fmt.Printf("%v %v %v\n", min, max, sum)
}

File: AOJ/ITP1/itp_1_4_d.cpp
#include <bits/stdc++.h>
using namespace std;
int main() {
  int n;
  scanf("%d", &n);
  int tmp;
  scanf("%d", &tmp);
  cout << tmp;
  int sum = 0;
  int min = n == 0 ? 0 : 1000000;
  int max = n == 0 ? 0 : -1000000;
  for (int i = 0; i < n; i++) {
    scanf("%d", &tmp);
    sum += tmp;
    if (min > tmp) min = tmp;
    if (max < tmp) max = tmp;
  }
  printf("%d %d %d\n", min, max, sum);
}
